### PR TITLE
feat: Add types for react-native-phone-input

### PR DESCRIPTION
**Pull request by maael**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-native-phone-input/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+     */
+    onPressFlag?: () => void;
+    /**
+     * Custom countries list
+     */
+    countriesList?: ReadonlyArray<CountriesListItem>;
+}
+
+export default class ReactNativePhoneInput<TextComponentType extends React.ComponentType = typeof TextInput> extends React.Component<ReactNativePhoneInputProps<TextComponentType>> {
+  picker?: React.Ref<ThisType<ReactNativePhoneInput>>;
```
Comment (armanio123) [inline types/react-native-phone-input/index.d.ts]: Documentation doesn't show `picker` as a property.

File: types/react-native-phone-input/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+     * Function to be invoked when press on flag image
+     */
+    onPressFlag?: () => void;
+    /**
+     * Custom countries list
+     */
+    countriesList?: ReadonlyArray<CountriesListItem>;
+}
+
+export default class ReactNativePhoneInput<TextComponentType extends React.ComponentType = typeof TextInput> extends React.Component<ReactNativePhoneInputProps<TextComponentType>> {
```
Comment (armanio123) [inline types/react-native-phone-input/index.d.ts]: There's some missing properties:
`blur`, `onPressCancel`, `onPressConfirm`

**Review by armanio123 — CHANGES_REQUESTED** [review]
Leaving a couple of comments. Documentation might be outdated. Let me know if you want to merge as is.

File: types/react-native-phone-input/index.d.ts (hunk 1 above)
Comment (maael) [reply types/react-native-phone-input/index.d.ts]: It's set [here](https://github.com/thegamenicorus/react-native-phone-input/blob/master/lib/index.js#L216). I'm not too sure whether to include it here like this or not, I don't have much experience typing refs.

File: types/react-native-phone-input/index.d.ts (hunk 2 above)
Comment (maael) [reply types/react-native-phone-input/index.d.ts]: Added in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38937/commits/66080914b81ba107d7025918c1507e3b059f28d6

File: types/react-native-phone-input/index.d.ts (hunk 1 above)
Comment (armanio123) [reply types/react-native-phone-input/index.d.ts]: Thanks, this is the correct place.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**